src/templates/nulldata.js

```diff
@@ -14,15 +14,15 @@ function check (script) {
 check.toJSON = function () { return 'null data output' }
 
 function encode (data) {
-  typeforce(types.Buffer, data)
+  typeforce([types.Buffer], data)
 
-  return bscript.compile([OPS.OP_RETURN, data])
+  return bscript.compile([OPS.OP_RETURN].concat(data))
 }
 
 function decode (buffer) {
   typeforce(check, buffer)
 
-  return buffer.slice(2)
+  return bscript.decompile(buffer).slice(1)
```

Comment:
but this is very dangerous and will likely burn unsuspecting users who upgrade without reading the CHANGELOG closely